File: CPP/Day3/Armstrong.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isArmstrong(int num) {
    int originalNum = num, sum = 0, digits = 0;

    while (originalNum != 0) {
        digits++;
        originalNum /= 10;
    }

    originalNum = num;

    while (originalNum != 0) {
        int digit = originalNum % 10;
        sum += pow(digit, digits);
        originalNum /= 10;
    }

    return sum == num;
}

int main() {
    int number;

    cout << "Enter a number: ";
    cin >> number;

    if (isArmstrong(number)) {
        cout << number << " is an Armstrong number." << endl;
    } else {
        cout << number << " is not an Armstrong number." << endl;
    }

    return 0;
}

File: Assignments/20_OCT_2025/Armstrong.java
package OCT_20_2025;

import java.util.Scanner;

public class Armstrong {
	public static void main(String[] args) {
	     Scanner sc = new Scanner(System.in);
	     System.out.println("Enter the number to check Armstrong.");
	     int num = sc.nextInt();
	     int original = num;
	     int sum = 0;
	     
	     int digits = String.valueOf(num).length();
	     while(num != 0) {
	       int digit = num % 10;
	       sum += Math.pow(digit, digits);
	       num = num / 10;
	     }
	     
	     if(sum == original) {
	    	 System.out.println("Num is Armstrong number ..");
	     }else {
	    	 System.out.println("Number is Not Armstrong number..");
	     }

		}

}

// armstrong num ex    153 = (1e3 + 5e3 + 3e3) = (1 + 125 + 27 = 153)
